JavaScript/JavaScript.sublime-syntax

```diff
@@ -15,6 +15,22 @@ variables:
   func_lookahead: '\s*\b(async\s+)?function\b'
   arrow_func_lookahead: '\s*(\basync\s*)?([_$[:alpha:]][_$[:alnum:]]*|\(([^()]|\([^()]*\))*\))\s*=>'
 
+  method_name: >-
+    (?x)(?:
+      {{identifier}}
+      | '(?:[^\\']|\\.)*'
+      | "(?:[^\\"]|\\.)*"
+      | \[ {{identifier}} (?:\.{{identifier}})* \]
+    )
+
+  line_continuation_lookahead: >-
+    (?x)
+    (?! \+\+ | -- )
+    (?=
+      != |
+      [ -+*/% ><= &|^ \[( ;,.:? ]
+    )
```

Comment:
I think this is here to handle expressions split over multiple lines. e.g. if one has 
```javascript
x = 5
+ 3
```
that is one expression, but 
```javascript
x = 5
y
```
is two. 

So this lookahead determines whether the expression continues after the new line, or if it should be terminated (i.e. implicit semicolon).